lmdeploy/messages.py

```diff
@@ -93,21 +78,12 @@ def special_word_token_ids(words):
                 return indexes
             return None
 
-        return EngineGenerationConfig(
-            n=gen_config.n,
-            logprobs=gen_config.logprobs,
-            max_new_tokens=gen_config.max_new_tokens,
-            min_new_tokens=gen_config.min_new_tokens,
-            top_p=gen_config.top_p,
-            top_k=gen_config.top_k,
-            temperature=gen_config.temperature,
-            repetition_penalty=gen_config.repetition_penalty,
-            ignore_eos=gen_config.ignore_eos,
-            random_seed=gen_config.random_seed,
-            skip_special_tokens=gen_config.skip_special_tokens,
-            stop_words=special_word_token_ids(gen_config.stop_words),
-            bad_words=special_word_token_ids(gen_config.bad_words),
-            logits_processors=gen_config.logits_processors)
+        stop_words = special_word_token_ids(self.stop_words) or []
+        bad_words = special_word_token_ids(self.bad_words) or []
+        stop_words.extend(self.stop_words_ids or [])
+        bad_words.extend(self.bad_words_ids or [])
+        self.stop_words_ids = list(set(stop_words)) or None
```

Comment:
stop_words could be a list of list, which can be use `set`
```python
a = [[123], [456]]
b = set(a) # TypeError: unhashable type: 'list'
```

Reply:
Why stop_words could be a list of list?

Reply:
I tested the following case:
```python
gen_config = GenerationConfig(
    stop_words=chat_template.stop_words,
    stop_token_ids=[[92542], [92540]]  # list of list
)
```